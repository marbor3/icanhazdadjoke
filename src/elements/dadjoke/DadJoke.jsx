import react, { useEffect, useState } from "react";
import styles from "./DadJoke.module.css";

export default function DadJoke({ joke }) {
  return <article className={styles.article}>{joke}</article>;
}
