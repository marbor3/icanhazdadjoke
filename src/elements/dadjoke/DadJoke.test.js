import React from "react";
import { render } from "@testing-library/react";
import DadJoke from "./DadJoke";

describe("Elements - DadJoke", () => {
  test("renders given joke", async () => {
    const joke =
      "How come the stadium got hot after the game? Because all of the fans left.";
    const { getByTestId, container } = render(<DadJoke joke={joke} />);

    expect(getByTestId("joke")).toHaveTextContent(joke);
    expect(container).toMatchSnapshot();
  });
});
