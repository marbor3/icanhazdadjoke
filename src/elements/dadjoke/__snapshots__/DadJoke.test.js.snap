// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Elements - DadJoke renders given joke 1`] = `
<div>
  <article
    class="article"
    data-testid="joke"
  >
    How come the stadium got hot after the game? Because all of the fans left.
  </article>
</div>
`;
